pkg/cache/v2/snapshot.go

```diff
@@ -194,3 +218,44 @@ func (s *Snapshot) GetVersion(typeURL string) string {
 	}
 	return s.Resources[typ].Version
 }
+
+// GetVersionMap will construct a verison map off the current state of a snapshot
+func (s *Snapshot) GetVersionMap() (map[string]map[string]string, error) {
```

Comment:
`GetVersionMap()` is only called in tests as the current code stands. That's an good suggestion though, but it might not be worth moving that computation into the snapshot creation if it doesn't get utilized anywhere else